pkg/frontend/querymiddleware/codec.go

```diff
@@ -673,8 +673,8 @@ func (c prometheusCodec) EncodeLabelsQueryRequest(ctx context.Context, req Label
 		return nil, fmt.Errorf("unknown query result response format '%s'", c.preferredQueryResultResponseFormat)
 	}
 
-	if consistency, ok := api.ReadConsistencyFromContext(ctx); ok {
-		r.Header.Add(api.ReadConsistencyHeader, consistency)
+	if level, ok := api.ReadConsistencyFromContext(ctx); ok {
```

Comment:
Note to reviewers: we should propagate HTTP headers in `EncodeLabelsQueryRequest()` too, **but** this function is currently dead code (not used) and `LabelsQueryRequest` doesn't have headers yet, so I haven't added support in this PR but it's something I'm tracking as part of this work.